src/main/java/com/karankumar/bookproject/ui/book/BookForm.java

```diff
@@ -582,6 +583,23 @@ private void setComponentMinWidth(HasSize[] components) {
         }
     }
 
+    private SerializablePredicate<LocalDate> endDatePredicate() {
+        return endDate -> !(endDate != null && dateStartedReading.getValue() != null
+                && endDate.isBefore(dateStartedReading.getValue()));
+    }
+
+    private SerializablePredicate<LocalDate> datePredicate() {
+        return startDate -> !(startDate != null && startDate.isAfter(LocalDate.now()));
+    }
+
+    private SerializablePredicate<Integer> positiveNumberPredicate() {
+        return series -> (series == null || series > 0);
+    }
+
+    private SerializablePredicate<String> authorPredicate() {
+        return firstName -> (firstName != null && !firstName.isEmpty());
```

Comment:
Since you changed both the first and last name validators, the local lambda variable would be better as `authorName`